templates/pipelines/pipeline_builder.html

```diff
@@ -18,25 +18,86 @@
       top: 50%;
       transform: translate(0, -50%);
     }
+    .fullscreen-toggle {
+      transition: background-color 0.2s ease;
+      position: absolute;
+      top: 8px;
+      right: 8px;
+      padding: 8px;
+      background-color: #e5e7eb; /* bg-gray-200 */
+      border-radius: 4px;
+      z-index: 10;
+    }
+    .fullscreen-toggle:hover {
+      background-color: #d1d5db; /* hover:bg-gray-300 */
+    }
+    .fullscreen-toggle svg {
+      pointer-events: none;
+    }
+    :fullscreen #pipelines-container {
+      width: 100%;
+      height: 100%;
+    }
   </style>
 {% endblock page_head %}
 {% block breadcrumbs %}
 {% endblock %}
 {% block app %}
-  <div class="max-w-7xl mx-auto" id="pipelineBuilder">
+  <div class="flex-1 relative" id="pipelines-container">
+    <button class="fullscreen-toggle" title="Toggle Full Screen">
+      <svg class="w-5 h-5" fill="none" stroke="currentColor" viewBox="0 0 24 24">
```

Comment:
I'd suggest using the `fa-solid fa-expand` font awesome icon here and the `fa fa-compress` icon for the "shrink" one. I think it will fit a bit better with the rest of the look